sources/admin.py

```diff
@@ -252,123 +281,108 @@ def phone_number_secondary_semiprivate_display(self, obj):
-        in user is not the one who created the interaction, then remove
-        notes field for them
+        Given a request and an object, determine if we need to hide the contact data for the object. If
+        the object is None (doesn't exist yet), always return False (we don't need to hide anything).
+
+        Returns True if we need to hide the data, False if the user has permissions to see/edit the data.
         """
-        obj = Person.objects.get(id=object_id)
-        if obj.privacy_level == 'searchable' and obj.created_by != request.user:
-            self.fieldsets = (
-                ('Privacy', {
-                    'fields': ('privacy_level',)
-                }),
-                ('General info', {
-                    'fields': (
-                        'prefix',
-                        'pronouns',
-                        'name',
-                        'title',
-                        'industries',
-                        'organization',
-                        'website',
-                        'type_of_expert',
-                        'expertise',
-                    ),
-                }),
-                ('Contact info', {
-                    'fields': (
-                        'email_address_semiprivate_display',
-                        'phone_number_primary_semiprivate_display',
-                        'phone_number_secondary_semiprivate_display',
-                        'linkedin',
-                        'twitter',
-                        'skype',
-                    ),
-                }),
-                ('Location info', {
-                    'fields': (
-                        'timezone',
-                        'city',
-                        'state',
-                        'country',
-                    ),
-                }),
-                ('Advanced info', {
-                    'classes': ('collapse',),
-                    'fields': (
-                        'entry_method',
-                        'entry_type',
-                        'created_by',
-                        # 'last_updated_by',
-                        'updated',
-                    ),
-                }),
-            )
-            self.readonly_fields = [
-                'created_by',
-                'entry_method',
-                'entry_type',
-                'email_address_semiprivate_display',
-                'phone_number_primary_semiprivate_display',
-                'phone_number_secondary_semiprivate_display',
-                'updated',
-            ]
+        if not obj:
+            # If creating a new `Person`, give all permissions
+            return False
+        elif obj.created_by == request.user:
+            # users can always see their own `Person`s
+            return False
+        elif obj.privacy_level in ['searchable', 'private_individual']:
+            # if the `Person` is at all private, and the user didn't create the
```

Comment:
After being a devoted "all lowercase to start comments person," I'm starting to come around to capitalizing — especially in this case so it doesn't initially look like a commented out conditional statement.

```suggestion
            # If the `Person` is at all private, and the user didn't create the
```